### ISSUE TITLE
Add Alternative name concept to datasets

### ISSUE THREAD

**Issue by IsakNaslundBh** [BHoM/BHoM_Datasets#60]
#### Description:

This will probably need to be migrated to BHoM/Engine for implementation, but raising here to start as most relevant repo.

Would want to incorporate some system of alternative names that could be used to match items in the libraries. Useful when using the match method in the library, or calling it from the side of the code. Also useful when pulling items from packages that might use a slightly different convention than the BHoM.

The library engine is already able to igonore case and whitespaces, but it can only do so much.

As an example, for the UK_SteelSections:

- RHS 50x30x4.0

Could have the alternate forms:

- RHS 50x30x4
- HRHS 50x30x4.0 (for hot formed)
- HFRHS 50x30x4.0 (for hot formed)
- RHSH 50x30x4.0 (for hot formed)
- 50x30x4.0 RHS

as well as all the permutations of the above.

Could potentially add some `AlternativeNames` fragment or similar to the Data_oM to store the names on the objects. As an alternative, could set up some mapping system for the Library_Engine that can know for a particular dataset how it can re-map the name text to an alternate form.

**Comment by JosefTaylor** [comment]
Agreed on this. After asking around, I've learned there's significant opposition to switching from fractional names for HSS and Angle sections to decimals in our revit families. AISC (American steel) has two standard naming conventions, one for electronic data interchange, and one for the book/drawings/etc. which are decimal and fractional respectively, so we very much need to accommodate both names.
